src/main/java/net/minecraftforge/common/util/math/Size2i.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.util.math;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nonnegative;
+
+public final class Size2i
```

Comment:
Vec2i mimicking the vanilla names? Where does Size come from?

Reply:
Yeah it's trying to follow the mcp convention.
Dimensions and Size are basically the same thing (width x height), but dimension can get confused with world dimensions.

Reply:
i was saying that you should rename it to Vec2i
Size really doesn't mean anything but we have quite a few VecNT classes.

Reply:
I could create `Vec2i` or use the existing `javax.vecmath.Tuple2i` but then we get `x` and `y` instead of `width` and `height`.